amlb/datasets/fileutils.py

```diff
@@ -11,7 +11,22 @@
 
 log = logging.getLogger(__name__)
 
-SUPPORTED_SCHEMES = ("http", "https")
+SUPPORTED_SCHEMES = ("http", "https", "s3")
+
+
+def s3_path_to_bucket_prefix(s3_path):
+    s3_path_cleaned = s3_path.split('://', 1)[1]
+    bucket, prefix = s3_path_cleaned.split('/', 1)
+
+    return bucket, prefix
+
+
+def is_s3_url(path):
+    if type(path) != str:
+        return False
+    if (path[:2] == 's3') and ('://' in path[:6]):
```

Comment:
What is the one optional char we except to be able to be present in `path[2:6]`? If none, then please consider my suggestion.

Reply:
The optional character is to support both `s3://` and `s3a://` , where a stands for 'accelerated'. It is a different file loading mechanism, with different performance tradeoffs.

Reply:
@yinweisu I'd recommend undoing this change so we can continue to support s3a. I'm unsure how relevant it is anymore since I believe s3a was integrated into s3 by default now, but probably good to still have it.

Reply:
There are also other formats beyond s3a, including s3n: https://stackoverflow.com/questions/33356041/technically-what-is-the-difference-between-s3n-s3a-and-s3

Reply:
With the OO approach we are going to use, we might simply have something like this:
`scheme_handlers = dict(
    http: HttpHandler(),
    https: HttpHandler(),
    s3: S3Handler(),
    s3a: S3Handler(),
)`
This way we don't need `is_s3_url()` anymore

Reply:
How do we detect if it is an s3 url in the first place then?

for example, you could have a local path called `'s3a/local/data'`.

Reply:
The code is using urlparse to extract the scheme. I think for a local path `s3a/local/data`, the scheme would be empty